src/optimagic/optimization/multistart.py

```diff
@@ -12,20 +12,388 @@
+        )
+
+    if (
+        not isinstance(options.mixing_weight_bounds, tuple)
+        or len(options.mixing_weight_bounds) != 2
+        or not set(type(x) for x in options.mixing_weight_bounds) <= {int, float}
+    ):
+        raise InvalidMultistartError(
+            f"Invalid mixing weight bounds: {options.mixing_weight_bounds}. Mixing "
+            "weight bounds must be a tuple of two numbers."
+        )
+
+    if not isinstance(options.convergence_relative_params_tolerance, int | float):
+        raise InvalidMultistartError(
+            "Invalid relative params tolerance:"
+            f"{options.convergence_relative_params_tolerance}. Relative params "
+            "tolerance must be a number."
+        )
+
+    if (
+        not isinstance(options.convergence_max_discoveries, int | float)
+        or options.convergence_max_discoveries < 1
+    ):
+        raise InvalidMultistartError(
+            f"Invalid max discoveries: {options.convergence_max_discoveries}. Max "
+            "discoveries must be a positive integer or infinity."
+        )
+
+    if not isinstance(options.n_cores, int) or options.n_cores < 1:
+        raise InvalidMultistartError(
+            f"Invalid number of cores: {options.n_cores}. Number of cores "
+            "must be a positive integer."
+        )
+
+    if not callable(options.batch_evaluator) and options.batch_evaluator not in (
+        "joblib",
+        "pathos",
+    ):
+        raise InvalidMultistartError(
+            f"Invalid batch evaluator: {options.batch_evaluator}. Batch evaluator "
+            "must be a Callable or one of 'joblib' or 'pathos'."
+        )
+
+    if options.batch_size is not None and (
+        not isinstance(options.batch_size, int) or options.batch_size < options.n_cores
+    ):
+        raise InvalidMultistartError(
+            f"Invalid batch size: {options.batch_size}. Batch size "
+            "must be a positive integer larger than n_cores, or None."
+        )
+
+    if not isinstance(options.seed, int | np.random.Generator | None):
+        raise InvalidMultistartError(
+            f"Invalid seed: {options.seed}. Seed "
+            "must be an integer, a numpy random generator, or None."
+        )
+
+    if options.exploration_error_handling not in ("raise", "continue"):
+        raise InvalidMultistartError(
+            f"Invalid exploration error handling: {options.exploration_error_handling}."
+            " Exploration error handling must be 'raise' or 'continue'."
+        )
+
+    if options.optimization_error_handling not in ("raise", "continue"):
+        raise InvalidMultistartError(
+            "Invalid optimization error handling:"
+            f"{options.optimization_error_handling}. Optimization error handling must "
+            "be 'raise' or 'continue'."
+        )
+
+
+# Internal Options
+# ======================================================================================
+
+
+@dataclass
+class InternalMultistartOptions:
```

Comment:
As additional protection this should maybe have a `__post_init__` that does some checks. Overhead is not a concern for things that happen just once in an entire multistart optimization.